vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -629,6 +653,24 @@ def create_weights(self, layer: Module, num_experts: int, hidden_size: int,
             layer.w13_input_scale = None
             layer.w2_input_scale = None
 
+    def _gaudi_weight_wrapper(self, weight_loader):
+        """Wrapper for Gaudi weight conversion."""
+
+        def wrapper(*args, **kwargs):
+            # args[0] is parameter, args[1] is loaded_weight
+            # weights will be always in fp8, but scales will be in fp32,
+            # so we can detect it by dtype
+            loaded_weight = args[1]
+            if loaded_weight.dtype == torch.float8_e4m3fn:
+                loaded_weight.data = (loaded_weight.data.float() * 0.5).to(
```

Comment:
Works with `--weights-load-device cpu` only? If so, we must reset `--weights-load-device` if `VLLM_HPU_CONVERT_TO_FP8UZ=True`.

Reply:
No, at this point in the code tensor is on CPU anyway, so it doesn't matter what we set as `--weights-load-device`